src/group_theory/congruence.lean

```diff
@@ -809,14 +809,30 @@ noncomputable def quotient_ker_equiv_range (f : M →* P) : (ker f).quotient ≃
         ⟨λ x y h, ker_lift_injective f $ by rcases x; rcases y; injections,
          λ ⟨w, z, hz⟩, ⟨z, by rcases hz; rcases _x; refl⟩⟩ }
 
-/-- The first isomorphism theorem for monoids in the case of a surjective homomorphism. -/
+/-- The first isomorphism theorem for monoids in the case of a homomorphism with right inverse. -/
+@[to_additive "The first isomorphism theorem for `add_monoid`s in the case of a homomorphism
+with right inverse."]
+def quotient_ker_equiv_of_right_inverse (f : M →* P) (g : P → M)
+  (hf : function.right_inverse g f) :
+  (ker f).quotient ≃* P :=
+{ to_fun := ker_lift f,
+  inv_fun := coe ∘ g,
+  left_inv := λ x, ker_lift_injective _ (by rw [function.comp_app, ker_lift_mk, hf]),
+  right_inv := hf,
+  .. ker_lift f }
+
+/-- The first isomorphism theorem for monoids in the case of a surjective homomorphism.
+
+See `quotient_ker_equiv_of_right_inverse` for a computable version.
```

Comment:
Is this the fully-namespaced name? Doc-gen won't link it otherwise

Reply:
```suggestion
For a `computable` version, see `con.quotient_ker_equiv_of_right_inverse`.
```